lua/nvim-ts-autotag/internal.lua

```diff
@@ -288,12 +289,58 @@ local function find_child_tag_node(opt)
     return find_tag_node(opt)
 end
 
-local function check_close_tag()
+local function find_start_tag(current)
+    local ts_tag = get_ts_tag()
+    if not ts_tag then
+        return nil
+    end
+
+    if current:type() ~= 'ERROR' then
+        return nil
+    end
+
+    local parent = current:parent()
+
+    for node in parent:iter_children() do
```

Comment:
maybe you can reuse find_child_match function.

Reply:
Indeed I had not noticed that there was already a function that already existed with a similar behavior (even more robust).

Thank you for this remark, I added a new commit for this optimization and now the code repeats less ^^